### PR TITLE
Beginnings of wifi support for the Wio Terminal

### PR DESCRIPTION
**Pull request by twitchyliquid64**
- Methods on UART to control interrupts and read flags
 - Wifi controller implementing Seeed's eRPC protocol
 - Generic trait `erpc::RPC` to abstract implementation of specific RPCs
 - Implementations for reading the version, starting a scan, and enumerating results
 - Demo of the above implementations in the `wifi_scan` example
 - Fixing the `clock` example now that I bumped the version constraint on the `nom` crate beyond what a different dependency could handle

### CODE REVIEW

File: boards/wio_terminal/src/wifi/wifi_rpcs.rs [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+#[derive(Debug, Copy, Clone)]
+#[repr(u32)]
+pub enum BssType {
+    Infra = 0,
+    Adhoc = 1,
+    Any = 2,
+    Unknown = core::u32::MAX,
+}
+
+const WEP_ENABLED: u32 = 1;
```
Comment (sajattack) [inline boards/wio_terminal/src/wifi/wifi_rpcs.rs:91]: You could use https://crates.io/crates/bitflags for this

File: boards/wio_terminal/src/wifi/wifi_rpcs.rs [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+        }
+
+        let mut res = GenericArray::<ScanResult, N>::default();
+
+        //let (data, ssid) = SSID::parse(data)?;
+        for i in 0..N::to_usize() {
+            res[i] = unsafe { *((data.as_ptr() as *const ScanResult).offset(i as isize)) };
+        }
+
+        //let s: ScanResult = unsafe { core::mem::transmute::<[u8;
```
Comment (sajattack) [inline boards/wio_terminal/src/wifi/wifi_rpcs.rs:318]: There's some commented out code here. Did you forget or is it still a work in progress?

File: boards/wio_terminal/src/wifi/wifi_rpcs.rs (hunk 1 above)
Comment (twitchyliquid64) [reply boards/wio_terminal/src/wifi/wifi_rpcs.rs]: Done (but in different crate now)

File: boards/wio_terminal/src/wifi/wifi_rpcs.rs (hunk 2 above)
Comment (twitchyliquid64) [reply boards/wio_terminal/src/wifi/wifi_rpcs.rs]: Fixed (but in different crate now).

File: hal/src/common/thumbv7em/sercom/v1/uart.rs [hunk 3]
```diff
@@ -171,7 +171,6 @@ macro_rules! uart {
                             w.rxpo().bits(rxpo); // Uses pad 3 for rx
                             w.txpo().bits(txpo); // Uses pad 2 for tx (and pad 3 for xck)
 
-                            w.form().bits(0x00); // No parity
```
Comment (jessebraham) [inline hal/src/common/thumbv7em/sercom/v1/uart.rs:174]: Just out of curiosity, what's the reason for this line being removed?

Comment (twitchyliquid64) [reply hal/src/common/thumbv7em/sercom/v1/uart.rs:174]: I think I removed it when getting the UART working. Its safe either way - the UART is reset, and `0x00` (no parity) is the default value. I can put it back if we want to minimise the diff.

Comment (jessebraham) [reply hal/src/common/thumbv7em/sercom/v1/uart.rs:174]: It's fine to leave as-is if it's reset to this value anyways. As I said, just curious.

**Review by jessebraham — APPROVED** [review]
This looks pretty good to me, thank you so much! I'm just going to update my RTL8720 and verify the example works (which I'm sure it does), then I'll be happy to merge.

**Pull request merged**